test/e2e/framework/framework.go

```diff
@@ -834,6 +835,22 @@ func (f *Framework) LogEvents(t *testing.T, log *logrus.Entry, namespace string,
 		event := events.Items[i]
 		log.WithField("lastTimestamp", event.LastTimestamp).WithField("type", event.Type).WithField("reason", event.Reason).WithField("message", event.Message).Info("Event!")
 	}
+
+	ctx := context.Background()
+	// Accessor takes an arbitrary object pointer and returns meta.Interface https://pkg.go.dev/k8s.io/apimachinery/pkg/api/meta#Accessor
+	objMeta, err := meta.Accessor(objOrRef)
+	require.NoError(t, err, "error getting object metadata")
+	deployment, err := f.KubeClient.AppsV1().Deployments(namespace).Get(ctx, objMeta.GetName(), metav1.GetOptions{})
+	require.NoError(t, err, "error getting deployment")
```

Comment:
It's exactly failing here as per CI/CD error. It couldn't find the deployment.
Could you help me if this is the right way of getting deployments details?

Reply:
Maybe a silly question - why are you getting a Deployment? `LogEvents` works on any K8s object and is used all around the test suites, so you just put somthing singularly test specific in a function that is a test wide utility - probably not a good idea!

I'd suggest moving the extra logging into the flaky test itself.

Reply:
> Maybe a silly question - why are you getting a Deployment? `LogEvents` works on any K8s object and is used all around the test suites, so you just put somthing singularly test specific in a function that is a test wide utility - probably not a good idea!
> 
> I'd suggest moving the extra logging into the flaky test itself.

I think, I misunderstood statement mentioned over here: https://github.com/googleforgames/agones/pull/3616#discussion_r1490331895
Deployment is not needed. I have added the Log Events in the latest commit.